common/src/main/scala/common/util/TimeUtil.scala

```diff
@@ -0,0 +1,22 @@
+package common.util
+
+import java.time.format.DateTimeFormatter
+import java.time.{OffsetDateTime, ZoneOffset}
+
+object TimeUtil {
+  /**
+    * Instead of "one of" the valid ISO-8601 formats, standardize on this one:
+    * https://github.com/openjdk/jdk/blob/jdk8-b120/jdk/src/share/classes/java/time/OffsetDateTime.java#L1886
+    */
+  private val Iso8601MillisecondsFormat = DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSSXXXXX")
```

Comment:
Is `XXXXX` always a constant `Z` in the render? That seems a weird way of specifying it but sure.

FWIW http://userguide.icu-project.org/formatparse/datetime lists 3 possible output formats for `XXXXX` so I guess I'm still utterly perplexed by format strings 🤷‍♂

Reply:
Yes.

The `XXXXXX` is actually just a signal to use the constant `"Z"`, by entering this branch of code matching on one-to-six uppercase `'X'` characters, and then calling `appendOffset(…, "Z")`:

https://github.com/openjdk/jdk/blob/7217c26dc17110b3070975f7f52afde9a47d8c6a/jdk/src/share/classes/java/time/format/DateTimeFormatterBuilder.java#L1636-L1640

The second argument to `appendOffset()` is the "noOffsetText". Earlier the offset has already been removed via `.atZoneSameInstant(ZoneOffset.UTC)`.

As an alternate formatting example, using one-to-six lowercase `'x'` characters would have triggered this code branch, calling `appendOffset(…, zero)`:

https://github.com/openjdk/jdk/blob/7217c26dc17110b3070975f7f52afde9a47d8c6a/jdk/src/share/classes/java/time/format/DateTimeFormatterBuilder.java#L1641-L1646

TL;DR Could have used just `X` instead of `XXXXXX`, but wanted to copy the exact line from:
 https://github.com/openjdk/jdk/blob/7217c26dc17110b3070975f7f52afde9a47d8c6a/jdk/src/share/classes/java/time/OffsetDateTime.java#L1886